packages/flutter/test/widgets/form_test.dart

```diff
@@ -1126,4 +1126,58 @@ void main() {
     await tester.pump();
     expect(find.text('test_error'), findsNothing);
   });
+
+  testWidgets('AutovalidateMode.onFocusChange', (WidgetTester tester) async {
+    final GlobalKey<FormState> formKey = GlobalKey<FormState>();
+    String? errorText(String? value) => '$value/error';
+
+    Widget builder() {
+      return MaterialApp(
+        theme: ThemeData(),
+        home: MediaQuery(
+          data: const MediaQueryData(),
+          child: Directionality(
+            textDirection: TextDirection.ltr,
+            child: Center(
+              child: Form(
+                key: formKey,
+                autovalidateMode: AutovalidateMode.onFocusChange,
+                child: Material(
+                  child: Column(
+                    children: <Widget>[
+                      TextFormField(
+                        initialValue: 'bar',
+                        validator: errorText,
+                      ),
+                      TextFormField(
+                        initialValue: 'bar',
+                        validator: errorText,
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(builder());
+
+    // No error text is visible yet.
+    expect(find.text(errorText('foo')!), findsNothing);
+
+    // Enter text in the first TextFormField
+    await tester.enterText(find.byType(TextFormField).first, 'foo');
```

Comment:
Does focusing the field trigger a validation? I think you should add expectations for the error text after this pumpAndSettle to show what happens before the field is unfocused.

Reply:
It only triggers after the field is blurred.